storage/fanout.go

```diff
@@ -190,48 +215,54 @@ func (f *fanoutAppender) Rollback() (err error) {
 type mergeGenericQuerier struct {
 	mergeFunc genericSeriesMergeFunc
 
-	primaryQuerier genericQuerier
+	primaryQuerier int
 	queriers       []genericQuerier
 	failedQueriers map[genericQuerier]struct{}
 	setQuerierMap  map[genericSeriesSet]genericQuerier
 }
 
-// NewMergeQuerier returns a new Querier that merges results of chkQuerierSeries queriers.
-// NewMergeQuerier will return NoopQuerier if no queriers are passed to it
-// and will filter NoopQueriers from its arguments, in order to reduce overhead
-// when only one querier is passed.
-// The difference between primary and secondary is as follows: f the primaryQuerier returns an error, query fails.
-// For secondaries it just return warnings.
-func NewMergeQuerier(primaryQuerier Querier, queriers []Querier, mergeFunc VerticalSeriesMergeFunc) Querier {
+// NewMergeQuerier returns a new Querier that merges results of given slice of queriers.
+// NewMergeQuerier will return NoopQuerier if no queriers or noopQueriers are passed.
+// Function allows to optionally mark certain querier in the slice to be primary one. The difference between primary and
+// rest is as follows: if the primary querier returns an error, query fails. For secondaries it just return warnings.
+// Argument 'primaryQuerier' set to -1 means no primary querier.
+
+// In case of overlaps in the returned data by all queriers, mergeFunc will be used.
+func NewMergeQuerier(queriers []Querier, primaryQuerier int, mergeFunc VerticalSeriesMergeFunc) Querier {
```

Comment:
this seems like quite an odd API, why not just pass the primary querier separately from the secondary ones?

Reply:
We discussed it here: https://github.com/prometheus/prometheus/pull/7059#discussion_r400914446

I think at the end that's the plan, will update in ~2h/tomorrow (:

Reply:
Wonder if primary should be slice for consistency... Maybe for now as we don't need that, let's stick to one.

Reply:
Prometheus only has one local tsdb.